spec/controllers/reports/regions_controller_spec.rb

```diff
@@ -550,37 +573,163 @@
       sign_in(cvho.email_authentication)
 
       period = Period.month("July 2018")
-      expect(MonthlyDistrictDataService).to receive(:new).with(region, period, medications_dispensation_enabled: false).and_call_original
-      get :monthly_district_data_report,
-        params: {id: region.slug, report_scope: "district", format: "csv", period: period.value}
+
+      exporter = MonthlyDistrictData::Hypertension.new(
+        region: region,
+        period: period,
+        medications_dispensation_enabled: false
+      )
+      allow(MonthlyDistrictData::Hypertension).to receive(:new).and_return(exporter)
+      expect(MonthlyDistrictData::Exporter).to receive(:new).with(exporter: exporter).and_call_original
+      get :hypertension_monthly_district_data,
+        params: {id: region.slug, report_scope: "district", format: "csv", period: period.attributes}
     end
   end
 
-  describe "#monthly_state_data_report" do
+  describe "#diabetes_monthly_district_data" do
+    let(:facility_group) { create(:facility_group, organization: organization) }
+    let(:facility) { create(:facility, facility_group: facility_group) }
+    let(:region) { facility.region.district_region }
+
+    it "returns 401 when user is not authorized" do
+      facility
```

Comment:
Instead of  forcing the facility to be created like this, since you don't need a reference to the created object, you can create the objects in a before block instead of using lets.

Reply:
Right, will make the changes